airflow/providers/snowflake/decorators/snowpark.py

```diff
@@ -0,0 +1,143 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import TYPE_CHECKING, Callable, Dict, Optional, Sequence
+
+from airflow.decorators.base import DecoratedOperator, task_decorator_factory
+from airflow.operators.python import PythonOperator
+from airflow.providers.snowflake.operators.snowflake import get_db_hook
+
+if TYPE_CHECKING:
+    import snowflake
+
+    from airflow.decorators.base import TaskDecorator
+
+try:
+    import snowflake.snowpark  # noqa
+except ImportError as e:
+    # Snowpark is an optional feature and if imports are missing, it should be silently ignored
+    # As of Airflow 2.3  and above the operator can throw OptionalProviderFeatureException
```

Comment:
Nit

```suggestion
    # As of Airflow 2.3 and above the operator can throw AirflowOptionalProviderFeatureException.
```

Reply:
I made this package required but installed only for Python 3.8 because it would be difficult to integrate with our Docker image to have one extra package installed in a Docker image for Python 3.8 only.